storage/Devices/BcacheCsetImpl.cc

```diff
@@ -336,11 +336,17 @@ namespace storage
     void
     BcacheCset::Impl::do_deactivate() const
     {
-	string cmd_line = ECHO_BIN " 1 > " + quote(SYSFS_DIR "/fs/bcache/" + get_uuid() + "/stop");
+	// TODO handle multiple BlkDevices?
+
+	const BlkDevice* blk_device = get_blk_devices().front();
```

Comment:
What if this is empty? I.e. if there is no block device at all?
Even if you are sure that this should not happen, the code should not crash at this point.